### PR TITLE
DigitalOcean one click deploy

### PR DESCRIPTION
**Pull request by kylewu**
https://github.com/AugurProject/augur-node/issues/293

Not like heroku, i could not find one button deployment support from DO.
Only 3rd party service, like https://installer.71m.us, supports deploy in DO with app.yml configuration.
Please check `https://github.com/kylewu/augur-node` and there is one `Install on DigitalOcean` button on top of README.

NOTE
===
installer.71m.us is HTTPS, when it checks our droplet status, it sends request through http to ip of the newly created droplet. This request will be blocked because of security reason (see browser console for detail). 
one solution is to deploy `installer.71m.us` yourself (check its page for detail)

Verification
===
curl  46.101.77.91:9001
Hello World

`46.101.77.91` is my droplet

### CODE REVIEW

File: src/blockchain/sync-augur-node-with-blockchain.ts [hunk 1]
```diff
@@ -55,8 +55,22 @@ function monitorEthereumNodeHealth(augur: Augur) {
   }, 5000);
 }
 
+const getHttpUrl = (network: NetworkConfiguration) => {
```
Comment (pgebheim) [inline src/blockchain/sync-augur-node-with-blockchain.ts]: This isn't necessary.

NetworkConfiguration will read from the environment by default, so if you run without a NetworkConfiguration, `network.ws === process.env.ETHEREUM_HTTP`.

Just use ETHEREUM_HTTP and ETHEREUM_WS in place of the old ENDPOINT_HTTP and ENDPOINT_WS.

File: src/blockchain/sync-augur-node-with-blockchain.ts [hunk 2]
```diff
@@ -55,8 +55,22 @@ function monitorEthereumNodeHealth(augur: Augur) {
...
+const getWsUrl = (network: NetworkConfiguration) => {
+  if (process.env.ENDPOINT_WS) {
+    return process.env.ENDPOINT_WS;
+  }
+  return network.ws;
+};
+
 export function syncAugurNodeWithBlockchain(db: Knex, augur: Augur, network: NetworkConfiguration, uploadBlockNumbers: UploadBlockNumbers, callback: ErrorCallback): void {
-  augur.connect({ ethereumNode: { http: network.http, ws: network.ws }, startBlockStreamOnConnect: false }, (): void => {
+  augur.connect({ ethereumNode: { http: getHttpUrl(network), ws: getWsUrl(network) }, startBlockStreamOnConnect: false }, (): void => {
```
Comment (pgebheim) [inline src/blockchain/sync-augur-node-with-blockchain.ts]: Falls into the same category as above.

File: app.yml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+name: augur-node
```
Comment (pgebheim) [inline app.yml]: I'd encapsulate all this functionality in a script and put it in /scripts, see docket-entrypoint.sh.

Even better would be to have DO just run the docker image.

**Comment by pgebheim** [comment]
Closing without merging -- I am definitely open to a script that can run on DO if provided credentials or a more supported service for doing the deploys.

**Comment by kylewu** [comment]
@pgebheim thanks for your comments
do you prefer shell script, python script or node? I'm planning to create another pr

**Comment by sharkcrayon** [comment]
@kylewu Sorry for the delay in addressing this. We'll be getting back to you next week on the best way to move forward.

**Comment by petong** [comment]
@kylewu The original idea was that the bounty for this task be a pre-built application on Digital Ocean and not a shell/python/node script. See examples here: https://www.digitalocean.com/products/one-click-apps/

**Comment by kylewu** [comment]
@petong yes, this PR is not the script.

@everyone please kindly follow the conversation and try to understand what happend here. This is the first PR I created and it’s about the one button deployment while the other (later) is about the script.

I did this in the beginning, then I was told a script is preferable. After the script PR was created I was told one button deployment is required.

If you believe one button is what you need, please leave your review in this PR. It won’t take more than 5 minutes to check code.

**Comment by kylewu** [comment]
I’m leaving this conversation because I don’t have enough time to follow this. I hope someone could continue. Thanks for your time.

**Comment by pgebheim** [comment]
Closing this PR, but I've taken this script and adapted it to be a bit more robust:

https://github.com/AugurProject/augur-node/pull/589

Thanks for the contrib @kylewu

**Pull request closed without merging**